packages/frontend/src/presentation/screens/__tests__/cardSearchFilter.js

```diff
@@ -39,44 +52,135 @@ jest.mock('../../Navigator', () => {
+          colors: ['U', 'W'],
         },
         {
           id: 'id2',
-          name: 'chandraFlameFury',
+          name: "Chandra, Flame's Fury",
+          subTypes: ['Chandra'],
+          superTypes: ['Legendary'],
+          colorIdentities: ['R'],
+          colors: ['R'],
+        },
+        {
+          id: 'id3',
+          name: 'Teferi, Hero of Dominaria',
+          subTypes: ['Teferi'],
+          superTypes: ['Legendary'],
+          colorIdentities: ['W'],
+          colors: ['W'],
+        },
+      ],
+    };
+    const searchCardInput = getByPlaceholder(SEARCH_INPUT_PLACEHOLDER);
+    const searchButton = getByText(SEARCH_BUTTON_TEXT);
+
+    fireEvent.changeText(searchCardInput, SEARCH_INPUT);
+    fireEvent.press(searchButton);
+
+    expect(searchCardInput.props.value).toEqual(SEARCH_INPUT);
+    await waitForElement(() =>
+      expect(navigation.navigate).toHaveBeenCalledWith(
+        toHaveBeenCalledWith,
+        navigateParams,
+      ),
+    );
+  });
+
+  test('should search for cards and navigate to search result filtered by subtype and supertype', async () => {
+    const navigateParams = {
+      cardsFiltered: [
+        {
+          id: 'id3',
+          name: 'Teferi, Hero of Dominaria',
+          subTypes: ['Teferi'],
+          superTypes: ['Legendary'],
+          colorIdentities: ['W'],
+          colors: ['W'],
+        },
+      ],
+    };
+    const searchCardInput = getByPlaceholder(SEARCH_INPUT_PLACEHOLDER);
+    const subtypeInput = getByPlaceholder(SUBTYPE_INPUT_PLACEHOLDER);
+    const supertypeInput = getByPlaceholder(SUPERTYPE_INPUT_PLACEHOLDER);
+    const searchButton = getByText(SEARCH_BUTTON_TEXT);
+
+    fireEvent.changeText(searchCardInput, SEARCH_INPUT);
+    fireEvent.changeText(subtypeInput, SUBTYPE_INPUT);
+    fireEvent.changeText(supertypeInput, SUPERTYPE_INPUT);
+    fireEvent.press(searchButton);
+
+    expect(searchCardInput.props.value).toEqual(SEARCH_INPUT);
+    await waitForElement(() =>
+      expect(navigation.navigate).toHaveBeenCalledWith(
+        toHaveBeenCalledWith,
+        navigateParams,
+      ),
+    );
+  });
+
+  test('should search for cards and navigate to search result filtered by color', async () => {
+    const navigateParams = {
+      cardsFiltered: [
+        {
+          id: 'id1',
+          name: 'Ajani, Inspiring Leader',
+          subTypes: ['Ajani'],
+          superTypes: ['Legendary'],
+          colorIdentities: ['U', 'W'],
+          colors: ['U', 'W'],
         },
         {
           id: 'id3',
-          name: 'chandraFlameFury',
+          name: 'Teferi, Hero of Dominaria',
+          subTypes: ['Teferi'],
+          superTypes: ['Legendary'],
+          colorIdentities: ['W'],
+          colors: ['W'],
         },
       ],
     };
     const searchCardInput = getByPlaceholder(SEARCH_INPUT_PLACEHOLDER);
     const searchButton = getByText(SEARCH_BUTTON_TEXT);
+    const manaColorCheckbox = await waitForElement(() =>
```

Comment:
nao sei se o problema é no seletor mas nao to conseguindo fazer o test funcionar direito.

Reply:
Acho que tem algo errado com os testes:
![image](https://user-images.githubusercontent.com/823150/68529270-2d681a00-02fd-11ea-9dab-07c3a1ef61df.png)